src/python/pants/scm/change_calculator.py

```diff
@@ -142,12 +144,14 @@ def iter_changed_target_addresses(self, changed_request):
     if changed_request.include_dependees not in ('direct', 'transitive'):
       return
 
-    # For dependee finding, we need to parse all build files.
-    product_iter = (t
-                    for targets in self._scheduler.product_request(HydratedTargets, [DescendantAddresses('')])
+    # For dependee finding, we need to parse all build files to collect all structs. But we
+    # don't need to fully hydrate targets (ie, expand their source globs).
+    adaptor_iter = (t
+                    for targets in self._scheduler.product_request(HydratedStructs,
```

Comment:
I haven't followed this one... What's a `HydratedStruct`? What does it actually contain? Is this the equivalent of a `Payload` in the Task world? If so, maybe we can call it a `Payload`? Or a `PayloadWithUnresolvedGlobs`?

Reply:
Was about to answer inline, but then added more information below the line on https://github.com/pantsbuild/pants/issues/4535#issue-225489152 instead. Short answer: about 80% of this will need to change.

Reply:
Cool :) That's a very useful summary, thanks! Can you paste it into a comment in code somewhere (maybe `src/python/pants/engine/struct.py`) and reference it from the places that each of these types is defined, so that people can work out what's going on by reading the code, rather than having to know which tickets explain the code?

Reply:
I'll split the baby and leave TODOs referring to that ticket.

Reply:
I can live with that if it's short-term, but I definitely value being able to develop while offline and reading comments in code is useful :)